packages/client/src/components/OrderDataCompletion/OrderDataCompletion.tsx

```diff
@@ -13,82 +17,389 @@ import {
+  phoneNumber: '800500300',
+};
+
+const addresses = [
+  {
+    id: '1',
+    street: 'Słowackiego',
+    streetNumber: '15',
+    apartmentNumber: '1',
+    floor: '1',
+    city: 'Wrocław',
+    postcode: '00-000',
+  },
+
+  {
+    id: '2',
+    street: 'Sienkiewicza',
+    streetNumber: '5',
+    city: 'Wrocław',
+    postcode: '00-000',
+  },
+
+  {
+    id: '3',
+    street: 'Sienkiewicza',
+    streetNumber: '5',
+    city: 'Wrocław',
+    postcode: '00-000',
+  },
+];
+
+const addressToString = (address: AddressDto) => {
+  return `${address.street} ${address.streetNumber} ${address.apartmentNumber ? `/${address.apartmentNumber},` : ','} ${
+    address.floor ? `p. ${address.floor},` : ''
+  } ${address.postcode} ${address.city}`;
+};
+
 const OrderDataCompletion = () => {
+  const [name, setName] = useState(userPersonalInfo.name || '');
+  const [surname, setSurname] = useState(userPersonalInfo.surname || '');
+  const [email, setEmail] = useState('');
+  const [phoneNumber, setPhoneNumber] = useState(userPersonalInfo.phoneNumber || '');
+  const [street, setStreet] = useState('');
+  const [streetNumber, setStreetNumber] = useState('');
+  const [apartmentNumber, setApartmentNumber] = useState('');
+  const [floor, setFloor] = useState('');
+  const [postcode, setPostcode] = useState('');
+  const [city, setCity] = useState('');
+  const [deliveryHours, setDeliveryHours] = useState('');
+  const [clearAddressRadioButton, setClearAddressRadioButton] = useState(false);
+
+  const [nameError, setNameError] = useState(false);
+  const [surnameError, setSurnameError] = useState(false);
+  const [emailError, setEmailError] = useState(false);
+  const [phoneNumberError, setPhoneNumberError] = useState(false);
+  const [streetError, setStreetError] = useState(false);
+  const [streetNumberError, setStreetNumberError] = useState(false);
+  const [apartmentNumberError, setApartmentNumberError] = useState(false);
+  const [floorError, setFloorError] = useState(false);
+  const [postcodeError, setPostcodeError] = useState(false);
+  const [cityError, setCityError] = useState(false);
+  const [deliveryHoursError, setDeliveryHoursError] = useState(false);
+
+  const [nameErrorMessage, setNameErrorMessage] = useState('');
+  const [surnameErrorMessage, setSurnameErrorMessage] = useState('');
+  const [emailErrorMessage, setEmailErrorMessage] = useState('');
+  const [phoneNumberErrorMessage, setPhoneNumberErrorMessage] = useState('');
+  const [streetErrorMessage, setStreetErrorMessage] = useState('');
+  const [streetNumberErrorMessage, setStreetNumberErrorMessage] = useState('');
+  const [apartmentNumberErrorMessage, setApartmentNumberErrorMessage] = useState('');
+  const [floorErrorMessage, setFloorErrorMessage] = useState('');
+  const [postcodeErrorMessage, setPostcodeErrorMessage] = useState('');
+  const [cityErrorMessage, setCityErrorMessage] = useState('');
+  const [deliveryHoursErrorMessage, setDeliveryHoursErrorMessage] = useState('');
```

Comment:
Można się obyć bez jakoś tak 22 stanów tutaj, bo `xError` to pure function `xErrorMessage`, a `xErrorMessage` to za to pure function `x` (gdzie `x` to `name`, `surname` itd.). Więcej info jakie są wady obecnego rozwiazania: https://prateeksurana.me/blog/why-you-should-avoid-using-state-for-computed-properties/, albo jeszcze więcej googlując coś w stylu "react computed properties".

Na przykładzie `name`: możemy wyciągnąć `nameErrorMessage` z `name` i `nameError` z `nameErrorMessage`:
```TS
const [name, setName] = useState('');
const nameErrorMessage = name.match(/^[a-zA-ZĄĆĘŁŃÓŚŹŻąćęłńóśźż]{3,35}$/) ? null : 'Wpisz poprawne imię.';
const nameError = nameErrorMessage !== null;
```
Masz wtedy jedno "źródło prawdy" zamiast 3. Albo w ogóle nie robić `nameError` i uznać, że pusty tekst błędu to brak błędu:
```TSX
const [name, setName] = useState('');
const nameErrorMessage = name.match(/^[a-zA-ZĄĆĘŁŃÓŚŹŻąćęłńóśźż]{3,35}$/) ? '' : 'Wpisz poprawne imię.';
// ...
  value={name}
  error={nameErrorMessage !== ''}
  helperText={nameErrorMessage}
```

Jedyna regresja względem obecnego rozwiązania to to, że błędy wyświetlają się od razu a nie po pierwszym submicie, ale można dodać pojedynczy stan do tego dodatkowy:
```TS
const [didSubmit, setDidSubmit] = useState(false);
```
Ustawić na `true` na submicie i wyświetlać błędy tylko kiedy jest `true`.

---

Wyleci kilkanaście linijek od `const [nameError, setNameError] = useState(false);` do `const [deliveryHoursErrorMessage, setDeliveryHoursErrorMessage] = useState('');`, kolejne kilkanaście od `setClearAddressRadioButton(false);` do `setCity(addressObject.city);`, wszystkie duplikaty w walidacji, kolejne kilkanaście od `setNameError(false);` do `setDeliveryHoursError(false);` i generalnie sumarycznie chyba ze sto linii (a do tego trzykrotne zmniejszenie liczby hooków, których jest teraz troszkę sporo).

---

Podobnie z formami do loginu/rejestracji, ale tam nie wspominałem, bo przy 3 inputach to tak nie boli jak 11 tutaj xD

Reply:
100x lepiej, dzięki! 😍